modules/ROOT/pages/configure-multiple-credential-providers.adoc

```diff
@@ -0,0 +1,116 @@
+
+If an API has only applied the Client ID Enforcement policy and not applied the OAuth policy, the client application is still created in an external IdP.
+When an OAuth policy is deployed to API gateway, the platform automatically injects IdP configuration, such as the token URL and token introspection url in the policy configuration. The IdP configuration is completely hidden from a user in API Manager.
+
+If no external IdPs are enabled, Access Manager acts as the default authorization server in that case. 
+//[PA:-Tomas not sure what the following statement is trying to say.]
+// OOTB OAuth policies with an external IdP can’t be applied 
+// ClientId enforcement policy can be applied.
+
+
+== The Use Case
+
+To better understand how to implement the multiple external credential provider feature, let us look at the example of "Good Weather" company that is in the business of forecasting weather for a given location and time. Good Weather collects quantitative data about the current state of the atmosphere at a given place and, using meteorology, predicts how that current state will change over time.
+
+Good Weather has several sub-organizations running under the master organization. These sub-organizations exist based on the functions they perform, for example, Engineering and Forecasting. The Forecasting department researches and collects recent meterological data from various sources. This data is then consumed by various client applications. While the Engineering department maintains the applications that host the forecasting research data.
+
+Good Weather needs to ensure that the forecasting data is accurate at all times. For that reason the weather APIs must be accessed only by employees of the Forecasting sub-organization. Good Weather also wants to ensure that the production APIs are the single source of truth at all times.
+
+To ensure security and segregation of access for these different organizations, environments, and APIs, Good Weather IT sets up a separate external client providers for the Forecasting sub-organization.
+
+//example of how they implement the client providers.
+//Tomas how will they best implement this?
+
+== Guidelines for Implementing Multiple Client Providers
+
+Before you implement multiple client providers in Anypoint Platform, ensure that you carefully review the following guidelines and best practices:
+
+* When changing the client provider for an existing API instance, ensure that the API:
+** Has no contracts
+** Has no client provider-related policies applied, such as PingFederate and OpenID policies.
+* When promoting and importing an API:
+
+* If you assign an external client provider to an environment, all existing APIs in that environment display Anypoint Platform in the UI by default. 
```

Comment:
I had commented on the doc. If you didn't an IDP assigned to your environment, after you assign an external IDP to it, all EXISTING APIs will still be using Anypoint. You can then reassign the required external IdP for that API.

Reply:
Done! Thanks!